library/build.gradle

```diff
@@ -15,6 +15,8 @@ android {
 dependencies {
   api "com.android.support:recyclerview-v7:$supportlib_version"
   implementation "com.android.support:appcompat-v7:$supportlib_version"
+  implementation "android.arch.lifecycle:extensions:$architecture_version"
+  annotationProcessor "android.arch.lifecycle:compiler:$architecture_version"
```

Comment:
Are you really using lifecycle classes?
I think ViewModel can work without the annotationProcessor

and I'm wondering if we should use retained Fragment rather than depending on Architecture Components(ViewModel is using retained Fragment to survive configuration change), to reduce the library's size

Reply:
Oops, I keep forgetting to remove those. Now that they're gone, is this PR good to merge? (I'd rather have that discussion on the relevant PR. 😄)

Reply:
OK, let's merge this!